app/screens/create_or_edit_channel/index.tsx

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, useEffect, useReducer} from 'react';
+import {useIntl} from 'react-intl';
+import {
+    Keyboard,
+    InteractionManager,
+} from 'react-native';
+import {Navigation} from 'react-native-navigation';
+
+import {switchToChannel} from '@actions/local/channel';
+import {handlePatchChannel, handleCreateChannel} from '@actions/remote/channel';
+import EditChannelInfo from '@components/edit_channel_info';
+import {General} from '@constants';
+import {useTheme} from '@context/theme';
+import {useFormInput} from '@hooks/forms';
+import {popTopScreen, dismissModal, setButtons} from '@screens/navigation';
+import {validateDisplayName} from '@utils/channel';
+
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type ChannelInfoModel from '@typings/database/models/servers/channel_info';
+
+type Props = {
+        serverUrl: string;
+        componentId: string;
+        channel?: ChannelModel;
+        channelInfo?: ChannelInfoModel;
+}
+
+const CLOSE_CHANNEL_ID = 'close-channel';
+const EDIT_CHANNEL_ID = 'update-channel';
+const CREATE_CHANNEL_ID = 'create-channel';
+
+type Button = {
+    testID: string;
+    id: string;
+    enabled: boolean;
+    showAsAction: string;
+    color: string;
+    text: string;
+};
+
+enum RequestActions {
+  START = 'Start',
+  COMPLETE = 'Complete',
+  FAILURE = 'Failure',
+}
+
+interface RequestState {
+    error: string;
+    saving: boolean;
+    type: ChannelType;
+}
+
+interface RequestAction {
+    type: RequestActions;
+    error?: string;
+}
+
+const CreateOrEditChannel = ({serverUrl, componentId, channel, channelInfo}: Props) => {
+    const intl = useIntl();
+    const {formatMessage} = intl;
+    const theme = useTheme();
+
+    const [type, setType] = useState<ChannelType>(channel?.type as ChannelType || General.OPEN_CHANNEL);
+    const [rightButton, setRightButton] = useState<Button>();
+
+    const displayName = useFormInput(channel?.displayName);
+    const purpose = useFormInput(channelInfo?.purpose);
+    const header = useFormInput(channelInfo?.header);
+
+    const emitCanSaveChannel = (enabled: boolean) => {
+        setButtons(componentId, {
+            leftButtons: [],
+            rightButtons: [{...rightButton, enabled}] as never[],
+        });
+    };
+
+    const isDirect = (): boolean => {
+        return channel?.type === General.DM_CHANNEL || channel?.type === General.GM_CHANNEL;
+    };
+
+    const close = (goBack = false): void => {
```

Comment:
On every render we create a new close reference. It should not be a big deal, since the last reference should be handled by the GC at some point. Nevertheless, I personally preferred to define this "not dependent on anything in the component" functions to be declared outside. That way, if they ever depend on something in the component, we have to move them inside and think about if any effect, callback, etc... is using them, so we have to add it to the dependencies.

Reply:
done.  I moved this ouside of the component.